lms/djangoapps/grades/tests/test_models.py

```diff
@@ -97,8 +97,15 @@ def test_creation(self):
         list_of_block_dicts = [self.record_a._asdict()]
         for block_dict in list_of_block_dicts:
             block_dict['locator'] = unicode(block_dict['locator'])  # BlockUsageLocator is not json-serializable
-        expected_json = json.dumps(list_of_block_dicts, separators=(',', ':'), sort_keys=True)
+        expected_data = {
+            'course_key': unicode(self.record_a.locator.course_key),
+            'blocks': [
+                {'locator': unicode(self.record_a.locator), 'max_score': 10, 'weight': 1},
+            ],
+        }
+        expected_json = json.dumps(expected_data, separators=(',', ':'), sort_keys=True)
         expected_hash = b64encode(sha1(expected_json).digest())
+        self.assertEqual(expected_data, json.loads(vblocks.blocks_json))
```

Comment:
Yeah.  I didn't like the fact that the test had the same logic that the actual code had.  Kinda takes away the point of the test.